lib/runners/qemu.nix

```diff
@@ -341,19 +341,20 @@ lib.warnIf (mem == 2048) ''
     else throw "Cannot shutdown without socket";
 
   setBalloonScript =
-    if socket != null && balloon
-    then ''
+    if balloon
+    then
+      if socket != null then
+      ''
       VALUE=$(( $SIZE * 1024 * 1024 ))
       SIZE=$( (
-        ${writeQmp { execute = "qmp_capabilities"; }}
         ${writeQmp { execute = "balloon"; arguments.value = 987; }}
       ) | sed -e s/987/$VALUE/ | \
         ${pkgs.socat}/bin/socat STDIO UNIX:${socket},shut-none | \
-        tail -n 1 | \
```

Comment:
How did you test that this did not break anything?

Reply:
That's because I removed the echo capabilities line, which doesn't do anything because it gets tailed away. That line doesn't seem to do anything at all because the user won't be able to see the output.